### PR TITLE
{tools}[foss/2021a] PuLP v2.5.1 w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by manifestoso**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by manifestoso** [comment]
Test report by @robqiao
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
hpc-head01 - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, Python 3.9.6
See https://gist.github.com/d3d5771a1edc509b36e97330e5f7c703 for a full test report.

File: easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Contribution by
+# DeepThought, Flinders University
+# R.QIAO <rob.qiao@flinders.edu.au>
+
+easyblock = 'PythonPackage'
+
+name = 'python-pulp'
```
Comment (lexming) [inline easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb:7]: Based on its README file, the name of this software should be PuLP. We only prepend software names with *python* in cases where the raw name is already in use. This usually happens with python bindings to other software.
```suggestion
name = 'PuLP'
```

File: easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+dependencies = [
+    ('Python', '3.9.5'),
+]
+
+download_dep_fail = True
+use_pip = True
+
+sanity_pip_check = True
+
+options = {'modulename': 'pulp'}
```
Comment (lexming) [inline easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb:33]: This won't be necessary once the name is changed to PuLP

File: easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+toolchain = {'name': 'foss', 'version': '2021a'}
+toolchainopts = {'usempi': True}
+
+source_urls = ['https://pypi.python.org/packages/source/%(nameletter)s/pulp']
+sources = ['PuLP-%(version)s.tar.gz']
+checksums = ['27c2a87a98ea0e9a08c7c46e6df47d6d4e753ad9991fea2901892425d89c99a6']
+
+dependencies = [
+    ('Python', '3.9.5'),
+]
```
Comment (lexming) [inline easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb:26]: We can add dependencies to Cbc and GLPK, which are easily available and already have easyconfigs. I would also add a dependency on Gurobi, but leave it commented out as it requires a specific license.

**Review by lexming — CHANGES_REQUESTED** [review]
Thanks for your contribution @manifestoso . Please check my comments below about the name and dependencies of this easyconfig

File: easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb (hunk 1 above)
Comment (manifestoso) [reply easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb]: > Based on its README file, the name of this software should be PuLP. We only prepend software names with _python_ in cases where the raw name is already in use. This usually happens with python bindings to other software.

@lexming `python` packages are many and clog the directory with simple `python` packages are difficult to visualize what packages are still required for maintenance. A prefix, I think would be a good idea, i.e. `spack` adopts such a convention.

File: easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb (hunk 3 above)
Comment (manifestoso) [reply easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb:26]: @lexming good spot, thx.

Comment (manifestoso) [reply easybuild/easyconfigs/p/python-pulp/python-pulp-2.5.1-foss-2021a.eb:26]: > We can add dependencies to Cbc and GLPK, which are easily available and already have easyconfigs. I would also add a dependency on Gurobi, but leave it commented out as it requires a specific license.

@lexming done.

**Comment by manifestoso** [comment]
Test report by @robqiao
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
hpc-head01 - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, Python 3.9.6
See https://gist.github.com/3ca8347a2b66e73d1206c1a55a042885 for a full test report.

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node301.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/06769c8fae764fc8155d6c66cac85037 for a full test report.

**Review by lexming — APPROVED** [review]
Thanks for the update @manifestoso . I made a few additional changes:
* removed `usempi` as PuLP does not seem to use it
* use their GH repo as homepage instead of the page in PyPI
* remove `sources_url` as any `PythonPackage` by default will fetch its sources from PyPI

Let me know if you agree with these changes.

LGTM

**Comment by lexming** [comment]
Going in, thanks @manifestoso !

**Pull request merged**